components/patient-data/impl/src/main/java/org/phenotips/data/internal/PhenoTipsPatient.java

```diff
@@ -177,8 +176,12 @@ private void loadDisorders(XWikiDocument doc, BaseObject data) throws XWikiExcep
     private void loadSerializers()
     {
         try {
-            this.serializers =
-                ComponentManagerRegistry.getContextComponentManager().getInstanceList(PatientDataController.class);
+            List<PatientDataController<?>> availableSerializers = ComponentManagerRegistry
+                    .getContextComponentManager()
+                    .getInstanceList(PatientDataController.class);
+            for (PatientDataController<?> serializer : availableSerializers) {
+                this.serializers.put(serializer.getName(), serializer);
```

Comment:
I'd suggest logging a warning if two serializers have the same name instead of silently overwriting the first one.